mm2src/lp_ordermatch.rs

```diff
@@ -3285,6 +3360,129 @@ pub async fn set_price(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, Strin
     Ok(try_s!(Response::builder().body(res)))
 }
 
+pub async fn update_maker_order(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, String> {
+    let req: MakerOrderUpdateReq = try_s!(json::from_value(req));
+
+    let ordermatch_ctx = try_s!(OrdermatchContext::from_ctx(&ctx));
+    let my_maker_orders = ordermatch_ctx.my_maker_orders.lock().await;
+
+    let (base_coin, rel_coin, original_price, original_volume, updated_conf_settings, matches) =
+        match my_maker_orders.get(&req.uuid) {
+            Some(order) => {
+                let base = order.base.as_str();
+                let base_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, base).await) {
+                    Some(coin) => coin,
+                    None => return ERR!("Base coin {} has been removed from config", base),
+                };
+
+                let rel = order.rel.as_str();
+                let rel_coin: MmCoinEnum = match try_s!(lp_coinfind(&ctx, rel).await) {
+                    Some(coin) => coin,
+                    None => return ERR!("Rel coin {} has been removed from config", rel),
+                };
+
+                let original_conf_settings = order.conf_settings.unwrap();
+
+                let updated_conf_settings = OrderConfirmationsSettings {
+                    base_confs: req.base_confs.unwrap_or(original_conf_settings.base_confs),
+                    base_nota: req.base_nota.unwrap_or(original_conf_settings.base_nota),
+                    rel_confs: req.rel_confs.unwrap_or(original_conf_settings.rel_confs),
+                    rel_nota: req.rel_nota.unwrap_or(original_conf_settings.rel_nota),
+                };
+
+                (
+                    base_coin,
+                    rel_coin,
+                    order.price.clone(),
+                    order.max_base_vol.clone(),
+                    updated_conf_settings,
+                    order.matches.clone(),
+                )
+            },
+            None => return ERR!("There is no order with UUID {}", req.uuid),
+        };
+
+    drop(my_maker_orders);
+
+    let mut update_msg = new_protocol::MakerOrderUpdated::new(req.uuid);
+    update_msg = update_msg.with_new_conf_settings(updated_conf_settings);
+
+    // Validate and Add new_price to update_msg if new_price is found in the request
+    let new_price = match req.new_price {
+        Some(new_price) => {
+            try_s!(validate_price(new_price.clone()));
+            update_msg = update_msg.with_new_price(new_price.clone().into());
+            new_price
+        },
+        None => original_price,
```

Comment:
Oh i see, in the other case we use original_price, but i still don't understand in which situation i would like to update a maker order without modifying the price ? I guess it's some specific cases such as modifying min_volume or nota settings let's say, am i correct in this case ?

Reply:
> Oh i see, in the other case we use original_price, but i still don't understand in which situation i would like to update a maker order without modifying the price ? I guess it's some specific cases such as modifying min_volume or nota settings let's say, am i correct in this case ?

I think it's quite interesting because with this new rpc call i can:

- Modify a price of a maker order
- Modify a setting of a maker order without modifying the price (keeping same price, but other settings)
- Modify a maker order to use max balance without specifying a new price

Reply:
oh you found it here 💯